ocaml-lsp-server/src/hover_req.ml

```diff
@@ -12,33 +12,63 @@ let environment_mode =
   | Some true -> Extended_variable
   | Some false | None -> Default
 
-let format_contents ~syntax ~markdown ~typ ~doc =
+let print_dividers sections =
+  match sections with
+  | [] -> ""
+  | _ ->
+    String.concat ~sep:"\n\n --- \n\n" sections
+
+let format_as_code_block strings =
+  match strings with
+  | [] -> ""
+  | _ -> sprintf "```\n %s \n```" (String.concat ~sep:" " strings)
+
+
+let format_contents ~syntax ~markdown ~typ ~doc ~(syntax_documentation:[`Found of Query_protocol.syntax_doc_result
+| `No_documentation ]) ~syntax_doc_setting =
```

Comment:
This line will be reflect multiple other changes I suggested:
- We can call the argument `syntax_doc` here and everywhere else, it's shorter and mimic `doc`.
- There is no need to annotate the argument with it's type, especially when this type will simply be a `syntax_doc_result option`
- I think there it's too late to check if the feature is enabled: we don't want Merlin to perform extra work if it's not going to be displayed in the end.